src/Modal/index.ts

```diff
@@ -0,0 +1,123 @@
+import {Component} from 'react';
+import {render} from 'react-universal-interface';
+import FocusLock from 'react-focus-lock';
+import {h, isClient, on, off, noop} from '../util';
+import {Overlay, IOverlayProps} from '../Overlay';
+
+let id = 0;
+
+const ESC = 27;
+
+export interface IModalProps extends IOverlayProps {
+  onEsc?: (event: KeyboardEvent) => void;
+}
+
+export interface IModalState {
```

Comment:
do we really need a sate?

Reply:
We need something to hold data, since we have state, we might already use it. Also it is a pattern in all other components of injecting state into children, [using lib I created](https://github.com/streamich/react-universal-interface).